index.js

```diff
@@ -143,12 +143,17 @@ exports.decorateTerms = (Terms, { React }) => {
           if (this.state.isLoading) return;
           if (!this.state.isRecording) {
             this.titleElement = document.querySelector('header');
-            recording.startRecording(this.state.canvases, this.fileName);
+            this.setState({ isRecording: true });
+
+            recording.startRecording(this.state.canvases, (blob, extension) => {
+              this.fileName = this.fileName + '.' + extension;
+              recording.save(this.fileName, blob);
+              this.setState({ isRecording: false });
```

Comment:
Is this function a callback because of the `this.SetState`? If why not put the `SetState` after the function? This way we can have a more linear order to our code by passing a str to `startRecording` instead of a callback.

Reply:
Hmm... let me reconsider the way I do this then